core/rawdb/schema.go

```diff
@@ -295,22 +303,70 @@ func addressUtxosKey(address string) []byte {
 	return append(AddressUtxosPrefix, address[:]...)
 }
 
+var UtxoKeyLength = len(UtxoPrefix) + common.HashLength + 2
+
 // This can be optimized via VLQ encoding as btcd has done
 // this key is 36 bytes long and can probably be reduced to 32 bytes
 func UtxoKey(hash common.Hash, index uint16) []byte {
 	indexBytes := make([]byte, 2)
 	binary.BigEndian.PutUint16(indexBytes, index)
-	return append(utxoPrefix, append(hash.Bytes(), indexBytes...)...)
+	return append(UtxoPrefix, append(hash.Bytes(), indexBytes...)...)
+}
+
+var UtxoKeyWithDenominationLength = len(UtxoPrefix) + common.HashLength + 3
+var PrunedUtxoKeyWithDenominationLength = 9
+
+func UtxoKeyWithDenomination(hash common.Hash, index uint16, denomination uint8) []byte {
+	indexBytes := make([]byte, 2)
+	binary.BigEndian.PutUint16(indexBytes, index)
+	return append(UtxoPrefix, append(hash.Bytes(), append(indexBytes, denomination)...)...)
+}
+
+func ReverseUtxoKey(key []byte) (common.Hash, uint16, error) {
```

Comment:
what is the need for this?

Reply:
A UTXO key is to hash+index. This helper function takes in a key and returns the reverse.